crates/subspace-farmer/src/farming.rs

```diff
@@ -61,13 +70,19 @@ impl Farming {
 }
 
 impl Drop for Farming {
+    /// by sending an empty message to the receiver, we can interrupt the process
     fn drop(&mut self) {
-        // we don't have to do anything in here
-        // this is for clarity and verbosity
+        // we actually don't have to do anything in here
+        // the default implementation will drop the sender automatically (without writing any code
+        // in here) and receiver will know that sender is dropped via an error
+        // we could handle that error, and achieve the same functionality (we actually do that too)
+        // but we choose to send an empty message instead, for clarity
         let _ = self.sender.take().unwrap().send(());
     }
```

Comment:
This is way shorter and delivers the same amount of information to me:
```rust
    fn drop(&mut self) {
        let _ = self.stop_sender.take().unwrap().send(());
    }
```

I don't see why it is necessary to write so much text for such a simple thing when you can rename a variable to make it obvious.